### PR TITLE
Add skeleton assembly logging

### PR DESCRIPTION
**Pull request by jasonmalinowski**
This pull request is "clean" pull request which adds various bits of logging so we can better track down the skeleton assembly issue we're currently seeing. Unlike this previous PRs, this one is actually safe to check in and merge.

*Review:* @dotnet/roslyn-ide, @dotnet/roslyn-infrastructure

### CODE REVIEW

**Comment by brettfo** [comment]
There are a lot of commits here, did you mean to create this against `stabilization`?

**Comment by jasonmalinowski** [comment]
I meant to not screw up git rebase. Fixed.

**Comment by jmarolf** [comment]
:+1:

**Comment by Pilchie** [comment]
:+1:

File: src/Test/Utilities/Shared/Pe/BrokenStream.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+using System;
+using System.IO;
+
+namespace Roslyn.Test.Utilities
+{
+    internal class BrokenStream : Stream
+    {
+        public int BreakHow;
+
+        public override bool CanRead
```
Comment (CyrusNajmabadi) [inline src/Test/Utilities/Shared/Pe/BrokenStream.cs]: =>

**Comment by jasonmalinowski** [comment]
@dotnet/roslyn-compiler: this change now also is including the underlying message and stack trace in the error. For now while we identify the flaky issues, we'll need the full stack. I'm entirely OK with us rolling back the inclusion of the stack by 1.3.

File: src/EditorFeatures/Test2/Rename/RenameEngineTests.CSharpConflicts.vb [hunk 2]
```diff
@@ -3,13 +3,19 @@
 Imports Microsoft.CodeAnalysis.Rename.ConflictEngine
 
 Namespace Microsoft.CodeAnalysis.Editor.UnitTests.Rename
-    Partial Public Class RenameEngineTest
+    Partial Public Class RenameEngineTests
         Public Class CSharpConflicts
+            Private ReadOnly _outputHelper As Abstractions.ITestOutputHelper
+
+            Sub New(outputHelper As Abstractions.ITestOutputHelper)
```
Comment (jaredpar) [inline src/EditorFeatures/Test2/Rename/RenameEngineTests.CSharpConflicts.vb]: Visibility modifier

File: src/EditorFeatures/Test2/Rename/VisualBasic/EscapingTests.vb [hunk 3]
```diff
@@ -4,10 +4,16 @@ Imports Microsoft.CodeAnalysis.Rename.ConflictEngine
 
 Namespace Microsoft.CodeAnalysis.Editor.UnitTests.Rename.VisualBasic
     Public Class EscapingTests
+        Private ReadOnly _outputHelper As Abstractions.ITestOutputHelper
+
+        Sub New(outputHelper As Abstractions.ITestOutputHelper)
```
Comment (jaredpar) [inline src/EditorFeatures/Test2/Rename/VisualBasic/EscapingTests.vb]: visibility modifier

File: src/EditorFeatures/Test2/Rename/VisualBasic/ImplicitReferenceConflictTests.vb [hunk 4]
```diff
@@ -4,12 +4,17 @@ Imports Microsoft.CodeAnalysis.Rename.ConflictEngine
 
 Namespace Microsoft.CodeAnalysis.Editor.UnitTests.Rename.VisualBasic
     Public Class ImplicitReferenceConflictTests
+        Private ReadOnly _outputHelper As Abstractions.ITestOutputHelper
+
+        Sub New(outputHelper As Abstractions.ITestOutputHelper)
```
Comment (jaredpar) [inline src/EditorFeatures/Test2/Rename/VisualBasic/ImplicitReferenceConflictTests.vb]: Visibility modifier

File: src/EditorFeatures/Test2/Rename/VisualBasic/InvalidIdentifierTests.vb [hunk 5]
```diff
@@ -4,10 +4,16 @@ Imports Microsoft.CodeAnalysis.Rename.ConflictEngine
 
 Namespace Microsoft.CodeAnalysis.Editor.UnitTests.Rename.VisualBasic
     Public Class InvalidIdentifierTests
+        Private ReadOnly _outputHelper As Abstractions.ITestOutputHelper
+
+        Sub New(outputHelper As Abstractions.ITestOutputHelper)
```
Comment (jaredpar) [inline src/EditorFeatures/Test2/Rename/VisualBasic/InvalidIdentifierTests.vb]: visibility modifier

File: src/EditorFeatures/Test2/Rename/VisualBasic/QualificationTests.vb [hunk 6]
```diff
@@ -4,11 +4,17 @@ Imports Microsoft.CodeAnalysis.Rename.ConflictEngine
 
 Namespace Microsoft.CodeAnalysis.Editor.UnitTests.Rename.VisualBasic
     Public Class QualificationTests
+        Private ReadOnly _outputHelper As Abstractions.ITestOutputHelper
+
+        Sub New(outputHelper As Abstractions.ITestOutputHelper)
```
Comment (jaredpar) [inline src/EditorFeatures/Test2/Rename/VisualBasic/QualificationTests.vb]: Visibility modifier

File: src/Compilers/Core/Portable/Compilation/Compilation.cs [hunk 7]
```diff
@@ -1908,7 +1908,7 @@ private static EmitResult ToEmitResultAndFree(DiagnosticBag diagnostics, bool su
                 }
                 catch (Cci.PeWritingException e)
                 {
-                    diagnostics.Add(MessageProvider.CreateDiagnostic(MessageProvider.ERR_PeWritingFailure, Location.None, e.Message));
+                    diagnostics.Add(MessageProvider.CreateDiagnostic(MessageProvider.ERR_PeWritingFailure, Location.None, e.InnerException.ToString()));
```
Comment (jaredpar) [inline src/Compilers/Core/Portable/Compilation/Compilation.cs]: Can we add an assert to `PeWritingException` that we are given a non-null `InnerException` value?

File: src/Compilers/VisualBasic/Portable/VBResources.Designer.vb [hunk 8]
```diff
@@ -9010,6 +9010,15 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         End Property
         
         '''<summary>
+        '''  Looks up a localized string similar to An error occurred while writing the output file: {0}.
+        '''</summary>
+        Friend ReadOnly Property ERR_PeWritingFailure() As String
+            Get
+                Return ResourceManager.GetString("ERR_PeWritingFailure", resourceCulture)
+            End Get
+        End Property
```
Comment (jaredpar) [inline src/Compilers/VisualBasic/Portable/VBResources.Designer.vb]: Hmm, a bit surprised this was needed.  How was this code compiling before?  

Want to make sure we're not missing a case in our update 2 bits here. 

@TyOverby

**Comment by jaredpar** [comment]
LGTM modulo comments.  In particular I'm interested to know why that one piece of code is necessary.  That should have been necessary, to my reading, for the original change from @TyOverby to have occurred.  Want to make sure we didn't miss a case for update 2.

File: src/Compilers/VisualBasic/Portable/VBResources.Designer.vb (hunk 8 above)
Comment (jasonmalinowski) [inline src/Compilers/VisualBasic/Portable/VBResources.Designer.vb]: There were more arguments being passed than the string had placeholders for. Is there any validation of that?

Comment (jaredpar) [inline src/Compilers/VisualBasic/Portable/VBResources.Designer.vb]: Right but why did you need a new entry in the designer file?  You didn't add a new resource string, you modified the contents of an existing one.  That shouldn't effect the designer file.

**Comment by jasonmalinowski** [comment]
@jaredpar: just so you know, we'll be addressing the visibility modifier requests in a followup PR to avoid having to recycle Jenkins for that.

**Comment by CyrusNajmabadi** [comment]
:+1:
